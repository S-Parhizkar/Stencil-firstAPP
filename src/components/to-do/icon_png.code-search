# Query: icon.png

1 result - 1 file

src\manifest.json:
  7:     "src": "assets/icon/icon.jpg",
