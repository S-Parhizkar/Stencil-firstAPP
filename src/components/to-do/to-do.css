:host {
  display: block;
}

div {
  background:#ABDCD6;
  border-radius: 10px;
  padding-left:  15px;
  margin-bottom: 10px;
  font-family: cursive;
}
.outDiv{
  padding-top: 15px;
}
label {
  font-family: system-ui, sans-serif;
  font-size: 1rem;
  font-weight: bold;
  line-height: 1.1;
  margin: 10px 0;
  font-family: cursive;
}
span{
  color: #EB593C;
  font-weight: bold;
}
.delete {
  box-shadow: inset 0px 1px 0px 0px #EB593C;
  background: linear-gradient(to bottom, #EB593C 5%, #a8260c 50%);
  background-color: #EB593C;
  border-radius: 15px;
  border: 1px solid #942913;
  display: inline-block;
  cursor: pointer;
  color: #ffffff;
  font-family: cursive;
  font-size: 13px;
  padding: 6px 6px;
  text-decoration: none;
  text-shadow: 0px 1px 0px #854626;
  margin-left: 75%;
  margin-bottom: 10px;
  width: 60px;
}
.delete:hover {
  box-shadow: rgba(255, 255, 255, .2) 0 3px 15px inset, rgba(0, 0, 0, .1) 0 3px 5px, rgba(0, 0, 0, .1) 0 10px 13px;
  transform: scale(1.05);
}
.delete:active {
  position: relative;
  top: 1px;
}

.editbtn{
  box-shadow: inset 0px 1px 0px 0px #77eb94;
  background: linear-gradient(to bottom, #2bdb48 5%, #02640f 50%);
  background-color: #0aee3b;
    border-radius: 15px;
    border: 1px solid #6b1805;
    display: inline-block;
    cursor: pointer;
    color: #ffffff;
    font-family: cursive;
    font-size: 13px;
    padding: 6px 6px;
    text-decoration: none;
    text-shadow: 0px 1px 0px #854629;
    margin-top: 5px;
    width: 60px;
}
.editbtn:hover {
   box-shadow: rgba(255, 255, 255, .2) 0 3px 15px inset, rgba(0, 0, 0, .1) 0 3px 5px, rgba(0, 0, 0, .1) 0 10px 13px;
  transform: scale(1.05);
}