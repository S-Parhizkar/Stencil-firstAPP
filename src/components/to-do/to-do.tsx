import { Component, h, Prop } from '@stencil/core';
import { ITodo } from '../types';

@Component({
  tag: 'to-do',
  styleUrl: 'to-do.css',
  shadow: true,
})
export class ToDo {
  @Prop() todo: ITodo;

  handleChange() {
    this.todo.completed = !this.todo.completed
    fetch(this.todo.url, {
    method: 'PUT',
    body: JSON.stringify(this.todo)

    });
  }


 
  

  render() {
    return (
      <div>
        <div>Completed: {this.todo.completed ? 'YES' : 'NO'}</div>
        <div>ID: {this.todo.id}</div>
        <div>Order: {this.todo.order}</div>
        <div>Title: {this.todo.title}</div>
        <div>URL: {this.todo.url}</div>

        <form action="">
          <label>
            Check if completed:
            <input type="checkbox" checked={this.todo.completed} onClick={() => this.handleChange()} />
          </label>
        </form>

        <hr />
      </div>
    );
  }
}
