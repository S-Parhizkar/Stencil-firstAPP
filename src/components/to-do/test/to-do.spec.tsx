import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { ToDo } from '../to-do';

describe('to-do', () => {
  const todo: ITodo = {
    completed: true,
    id: 2,
    order: 3,
    title: 'sport',
    url: 'https://my-list-to-do.netlify.app/',
  };

  let page: SpecPage;

  beforeEach(async ()=> {
    page = await newSpecPage({
      components: [ToDo],
      template: () => <to-do todo={todo}></to-do>,
    });
    await page.waitForChanges();
  })

it('render', ()=>{
  expect(page.root).toMatchSnapshot();
})

it('should emit [update-todo] event to check once clicked on check nox ', async () => {
  const myMock = jest.fn();
  page.body.addEventListener('update-todo', myMock);
  (page.root.shadowRoot.querySelector('input#checkbox') as HTMLInputElement).click();
  await page.waitForChanges();
  expect(myMock).toBeTruthy()
  })

it('should emit [todo-to-update] event open [edit-tood] modal once clicked on edit button', async () => {
  const myMock = jest.fn();
  page.body.addEventListener('todo-to-update', myMock);
  (page.root.shadowRoot.querySelector('button.editbtn') as HTMLInputElement).click();
  await page.waitForChanges();
  expect(myMock).toHaveBeenCalledWith(expect.objectContaining({detail:todo}));
})

it('should emit [delete-todo] event to delete the todo once clicked on delete button', async () => {
  const myMock = jest.fn();
  page.body.addEventListener('delete-todo', myMock);
  (page.root.shadowRoot.querySelector('button.delete') as HTMLInputElement).click();
  await page.waitForChanges();
  expect(myMock).toBeNull();
})
 
});
