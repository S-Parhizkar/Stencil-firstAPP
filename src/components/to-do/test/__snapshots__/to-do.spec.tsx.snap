// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-do render 1`] = `
<to-do>
  <mock:shadow-root>
    <div class="outDiv">
      <div class="round">
        <span>
          TITLE:
        </span>
        sport
      </div>
      <div class="round">
        <span>
          Order:
        </span>
        3
      </div>
      <div class="round">
        <span>
          ID:
        </span>
        2
      </div>
      <div class="round">
        <span>
          Completed?:
        </span>
        YES, Done!
      </div>
      <div class="round">
        <label>
          <input checked="" id="checkbox" type="checkbox">
          Click here if DONE
        </label>
      </div>
      <hr>
      <dave class="container">
        <button class="editbtn">
          Edit
        </button>
        <button class="delete">
          Delete
        </button>
      </dave>
    </div>
  </mock:shadow-root>
</to-do>
`;
